template/src/pageComponents/signature-mint/ContractDemo.tsx

```diff
@@ -67,40 +61,43 @@ export default function SignatureMintDemo() {
   /**
    * Free Mint Contract Logic
    */
-  const { config: freeMintConfig } = usePrepareContractWrite({
+  const { data: freeMintConfig } = useSimulateContract({
     // TODO: the chainId should be dynamic
     address: contractAddress,
     abi: contract.abi,
     functionName: 'freeMint',
     args: address ? [address, debouncedSigValue] : undefined,
-    enabled: signature.length > 0,
+    query: {
+      enabled: signature.length > 0,
+    },
   });
-  const { write: freeMint } = useContractWrite(freeMintConfig);
+  const { writeContract: freeMint } = useWriteContract();
 
   /**
    * Paid Mint Contract Write Logic
    */
-  const { config: paidMintConfig } = usePrepareContractWrite({
+  const { data: paidMintConfig } = useSimulateContract({
     // TODO: the chainId should be dynamic
     address: contractAddress,
     abi: contract.abi,
     functionName: 'mint',
     args: [address],
     value: parseEther('0.0001'), // You should read the contract, however, setting this to value to prevent abuse.
   });
-  const { write: paidMint } = useContractWrite(paidMintConfig);
+  const { writeContract: paidMint } = useWriteContract();
 
-  const usedFreeMintResponse = useContractRead({
+  const usedFreeMintResponse = useReadContract({
     // TODO: the chainId should be dynamic
     address: contractAddress,
     abi: contract.abi,
     functionName: 'usedFreeMints',
     args: [address],
-    enabled: (address?.length ?? 0) > 0,
-    watch: true, // Watch for changes in the data and update the state if they use their free mint
-    cacheTime: 0,
-    staleTime: 0,
+    query: {
+      enabled: (address?.length ?? 0) > 0,
+      staleTime: 0,
```

Comment:
was it intentional to not add `gcTime` since previously it had `cacheTime ` and `gcTime` is a new equivalend in react-query 5

Reply:
I didn’t know that! Will update this - thank you!🙏